### PR TITLE
Removed github.com/kirillDanshin/letsencrypt dependecy

### PR DESCRIPTION
**Pull request by albenik**
Removed github.com/kirillDanshin/letsencrypt dependecy
Which has been only used for sel-figned certificate generation for localhost and was outdated

### CODE REVIEW

File: tls.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+)
+
+const localhost = "localhost"
+
+func selfSignedCertificate(clientHello *tls.ClientHelloInfo) (*tls.Certificate, error) {
+	if clientHello.ServerName != localhost {
+		return nil, fmt.Errorf("self-signed certificate for %q not permitted", clientHello.ServerName)
+	}
+
+	priv, err := rsa.GenerateKey(rand.Reader, 1024)
```
Comment (kirillDanshin) [inline tls.go]: how about something like this?
```suggestion
	priv, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
```

File: app.go [hunk 2]
```diff
@@ -13,7 +13,11 @@ import "time"
 
 // SetName for the server
 func (app *App) SetName(name string) {
-	app.name = name
+	if len(name) > 0 {
```
Comment (kirillDanshin) [inline app.go:17]: this check breaks ability to set `name` to `""`

**Review by kirillDanshin — CHANGES_REQUESTED** [review]

File: app.go (hunk 2 above)
Comment (albenik) [reply app.go:17]: Yep! But logic in old (removed) code also silently discards empty app.name :)

Comment (kirillDanshin) [reply app.go:17]: hehe, yes, my bad :) but I think if we here, we should fix it too :slightly_smiling_face:

Comment (albenik) [reply app.go:17]: `OptAppName` option added
`App.SetName()` fixed and marked as deprecated

**Comment by kirillDanshin** [comment]
@albenik I think we can bump minor version and release it. thanks

**Comment by albenik** [comment]
done

**Pull request merged**